Project Title Editor

JoshuaSchaefer opened the pull request:
Add an editor for a project title and add the ability to add a custom translation to the title.

---
### Your checklist for this pull request
Please review the [contributing guidelines](https://github.com/Catrobat/Catroweb-Symfony/blob/develop/.github/contributing.md) and [wiki pages](https://github.com/Catrobat/catroweb-Symfony/wiki/) of this repository.

- [ ] Include the name and id of the Jira ticket in the PR’s title eg.: `SHARE-666 The devils ticket`
- [ ] Choose the proper base branch (*develop*)
- [ ] Confirm that the changes follow the project’s coding guidelines
- [ ] Verify that the changes generate no warnings and errors 
- [ ] Verify to commit no other files than the intentionally changed ones
- [ ] Include reasonable and readable tests verifying the added or changed behavior
- [ ] Verify that all tests are passing (CI), if not please state the test cases in the [section](#Tests) below
- [ ] Perform a self-review of the changes
- [ ] Stick to the project’s git workflow (rebase and squash your commits)
- [ ] Verify that your changes do not have any conflicts with the base branch
- [ ] Put your ticket into the `Code Review` section in [Jira](https://jira.catrob.at/)
- [ ] Post a message in the *#catroweb* [Slack channel](https://catrobat.slack.com) and ask for a code reviewer
- [ ] Check that your pull request has been successfully deployed to https://web-test-1.catrob.at/

### Additional Description
There are also updates to limit the project name length to 50 characters and return a HTTP_UNPROCESSABLE_ENTITY if a request is made with greater than 50 characters. This was discussed during the following conversation.
https://github.com/Catrobat/Catroweb/pull/2450#discussion_r780930539 https://github.com/Catrobat/Catroweb/pull/2450#discussion_r783040511

### Tests - additional information
There are new tests similar to the project description and credits to test the edit function and to add a custom translation.

JoshuaSchaefer commented on templates/Program/program.html.twig:
```diff
@@ -246,15 +255,18 @@
   ></div>
   <div class="js-project-description-credits"
        data-project-id="{{ program.id }}"
+       data-max-name-length=10
```
This still needs to be updated to match what is on the backend

JoshuaSchaefer replied:
Updated now

danielzhang130 commented:
Can you squeeze in these 2 changes? Thanks

https://github.com/Catrobat/Catroweb/pull/2450#discussion_r780930539
https://github.com/Catrobat/Catroweb/pull/2450#discussion_r783040511

danielzhang130 commented on templates/Program/program.html.twig:
```diff
@@ -27,6 +27,15 @@
+          </span>
+        {% endif %}
+
         {% if not app.user or not my_program %}
```
I think this is the else branch for the new if, right?

JoshuaSchaefer replied:
Yup, should be updated 👍

danielzhang130 commented on config/routes/fos_js_routing.yaml:
```diff
@@ -1,6 +1,13 @@
     resource: "@FOSJsRoutingBundle/Resources/config/routing/routing.xml"
 
+# @deprecated
+edit_program_name:
```
Do we need this? Can you try without these lines? 
My understanding is that I've already defined this route here
https://github.com/danielzhang130/Catroweb/blob/edd9ae597e90653f1cd2ba2baee89796300d5336/src/Catrobat/Controller/Web/ProgramController.php#L335

JoshuaSchaefer replied:
I tried it without the lines and I got an error and it didn't go through. It looks like the other calls that use the route do the same as done here so I am guessing it is okay. It would be nice if we didn't have to specify this

JoshuaSchaefer commented on assets/js/components/ProjectTextEditor.js:
```diff
@@ -167,12 +167,13 @@ export function ProjectTextEditor (projectDescriptionCredits, defaultText, progr
       $.get(url, function (data) {
-        if (parseInt(data.statusCode) === 200) {
+        const responseCode = parseInt(data.statusCode)
+        // TODO: remove data once API updated
```
This needs to be updated once the API is updated

danielzhang130 commented on assets/js/project.js:
```diff
@@ -91,6 +92,25 @@ Program(
 )
 
+const nameEditorConfig = {
+  maxLength: $projectDescriptionCredits.data('max-length'),
```
max-name-length

JoshuaSchaefer replied:
Updated 👍

danielzhang130 commented:
> Can you squeeze in these 2 changes? Thanks
> 
> [Catrobat#2450 (comment)](https://github.com/Catrobat/Catroweb/pull/2450#discussion_r780930539) [Catrobat#2450 (comment)](https://github.com/Catrobat/Catroweb/pull/2450#discussion_r783040511)

Can you include these 2 changes? It's from the edit name api pr review.

This status code will also need to be changed because of this.
https://github.com/danielzhang130/Catroweb/blob/b472f4a8a3db1e07aadc4142086d7034895bb5ab/assets/js/components/ProjectTextEditor.js#L182

JoshuaSchaefer commented:
> > Can you squeeze in these 2 changes? Thanks
> > [Catrobat#2450 (comment)](https://github.com/Catrobat/Catroweb/pull/2450#discussion_r780930539) [Catrobat#2450 (comment)](https://github.com/Catrobat/Catroweb/pull/2450#discussion_r783040511)
> 
> Can you include these 2 changes? It's from the edit name api pr review.
> 
> This status code will also need to be changed because of this.
> 
> https://github.com/danielzhang130/Catroweb/blob/b472f4a8a3db1e07aadc4142086d7034895bb5ab/assets/js/components/ProjectTextEditor.js#L182

Yup, they should be in now.

Commit:
Project Title Editor
Add an editor for a project title and add the ability to add a custom translation to the title.
